kobo/apps/audit_log/models.py

```diff
@@ -335,13 +335,54 @@ def create_from_request(cls, request):
             'asset-file-list': cls.create_from_file_request,
             'asset-export-list': cls.create_from_export_request,
             'exporttask-list': cls.create_from_v1_export,
+            'asset-bulk': cls.create_from_bulk_request,
         }
         url_name = request.resolver_match.url_name
         method = url_name_to_action.get(url_name, None)
         if not method:
             return
         method(request)
 
+    @staticmethod
+    def create_from_bulk_request(request):
+        try:
+            payload = request._data['payload']
+            action = payload['action']
+            asset_uids = payload['asset_uids']
+        except KeyError:
+            return  # Incorrect payload
+
+        if type(asset_uids) is not list or len(asset_uids) == 0:  # Nothing to do
+            return
+
+        bulk_action_to_audit_action = {
+            'archive': AuditAction.ARCHIVE,
+            'unarchive': AuditAction.UNARCHIVE,
+        }
+        audit_action = bulk_action_to_audit_action.get(action)
+        if audit_action is None:
+            return  # Unsupported action
+
+        source = get_human_readable_client_user_agent(request)
+        client_ip = get_client_ip(request)
+
+        for asset_uid in asset_uids:
+            asset = Asset.all_objects.get(uid=asset_uid)
```

Comment:
we should do this in bulk before the loop to minimize db requests

Reply:
the problem with that is the latest_version property can only be obtained from a full asset object... so for now we are stuck with this. Should I put a TODO for this?

Reply:
using the function `Asset.optimize_queryset_for_list` for this